crypto/signature.go

```diff
@@ -144,18 +147,15 @@ func verifySignature(pubKeyEntries openpgp.EntityList, origText io.Reader, signa
 
 		_, err = signatureReader.Seek(0, io.SeekStart)
 		if err != nil {
-			return newSignatureFailed()
+			return nil, newSignatureFailed()
 		}
 
-		signer, err = openpgp.CheckDetachedSignatureAndHash(pubKeyEntries, origText, signatureReader, allowedHashes, config)
-		if err != nil {
-			return newSignatureFailed()
-		}
+		sig, signer, err = openpgp.VerifyDetachedSignatureAndHash(pubKeyEntries, origText, signatureReader, allowedHashes, config)
 	}
 
-	if signer == nil {
-		return newSignatureFailed()
+	if err != nil || sig == nil || signer == nil {
```

Comment:
Here we are swallowing the error (we were already doing it though), shouldn't we wrap it into the returned error ?

Reply:
Yeah, I'd tend to agree, but I'd do it in a separate PR, since it's not strictly speaking related to this go-crypto update